### PR TITLE
Adding instructions for configuring GitLab SSO

### PR DESCRIPTION
**Pull request by it33**

### CODE REVIEW

**Comment by it33** [comment]
Replaced by this PR in a better location: https://github.com/mattermost/platform/pull/534

File: gitlab-sso.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+The following steps can be used to configure Mattermost to use GitLab as a single-sign-on (SSO) service.
+
+1. Login to your GitLab account and go to the Applications section either in Profile Settings or Admin Area.
+2. Add a new application called "Mattermost" with the following as Redirect URIs:
+  * `<your-mattermost-url>/login/gitlab/complete` (example: http://localhost:8065/login/gitlab/complete)
+  * `<your-mattermost-url>/signup/gitlab/complete`
+
+3. Submit the application and copy the given _Id_ and _Secret_ into the appropriate _SSOSettings_ fields in config/config.json
+
+4. Also in config/config.json, set _Allow_ to `true` for the _gitlab_ section, leave _Scope_ blank and use the following for the endpoints:
```
Comment (maxlazio) [inline gitlab-sso.md]: @it33 can you tell me something about `Scope`? If it is not blank what value can it be and what for can it be used?

Comment (jwilander) [inline gitlab-sso.md]: Hi @maxlazio , `Scope` is defined in the OAuth2 spec here https://tools.ietf.org/html/rfc6749#section-3.3 . GitLab doesn't use `Scope` in it's OAuth2 service provider implementation, so it is left blank here. The reason we have it at all is for other systems we would like to use as OAuth 2 providers. For example, Google requires specifying a `Scope` for the access token that will control how much the requesting app gets access too.

Does that make sense?

Comment (maxlazio) [inline gitlab-sso.md]: @jwilander Thanks, makes sense.

**Comment by jwilander** [comment]
+1

**Comment by jwilander** [comment]
Closing as it's replaced by #534

**Pull request closed without merging**